### PR TITLE
3.2.x paginator having

### PR DESCRIPTION
**Pull request by Jurigag**
Hello!

* Type: bug fix | new feature
* Link to issue: https://github.com/phalcon/cphalcon/issues/12111

**In raising this pull request, I confirm the following (please check boxes):**

- [x] I have read and understood the [Contributing Guidelines](https://github.com/phalcon/cphalcon/blob/master/CONTRIBUTING.md)?
- [x] I have checked that another pull request for this purpose does not exist.
- [x] I wrote some tests for this PR.

Small description of change: This is adding way to properly calculate pages and items when using having in builder.

Thanks

### CODE REVIEW

**Comment by Jurigag** [comment]
rebased @sergeyklay

**Comment by Jurigag** [comment]
@sergeyklay appveyor timed out

File: phalcon/paginator/adapter/querybuilder.zep [hunk 1]
```diff
@@ -191,11 +220,20 @@ class QueryBuilder extends Adapter
...
+		 * If we have having perform native count on temp table
 		 */
-		let result = totalQuery->execute(),
-			row = result->getFirst(),
-			rowcount = row ? intval(row->rowcount) : 0,
-			totalPages = intval(ceil(rowcount / limit));
+		if hasHaving {
+		    let sql = totalQuery->getSql();
+		    let db = totalBuilder->getDI()->get("db");
+		    let row = db->fetchOne(sprintf("SELECT COUNT(*) as rowcount FROM (%s) as T1", sql["sql"]), Db::FETCH_ASSOC, sql["bind"]),
```
Comment (sergeyklay) [inline phalcon/paginator/adapter/querybuilder.zep]: sprintf?

**Review by sergeyklay — CHANGES_REQUESTED** [review]

File: phalcon/paginator/adapter/querybuilder.zep (hunk 1 above)
Comment (Jurigag) [inline phalcon/paginator/adapter/querybuilder.zep]: It looks cleaner imho than:

`"SELECT COUNT(*) as rowcount FROM (" . sql["sql"] . " as T1"`

Comment (sergeyklay) [inline phalcon/paginator/adapter/querybuilder.zep]: But slower

Comment (Jurigag) [inline phalcon/paginator/adapter/querybuilder.zep]: Oh right, thought it could be somehow optimized, yea i will change it i guess.

**Comment by Jurigag** [comment]
Removed sprintf @sergeyklay

**Comment by sergeyklay** [comment]
Thank you

**Pull request merged**